balance_stap.c

```diff
@@ -0,0 +1,40 @@
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+void balance(char* str)
+{
+    // коэф для проверки вложености скобок во время пробежки по строке(если коэф меньше 0, вложенность нарушена)
+    // открывающая скобка имеет коэф +1
+    // закрывающая скобка имеет коэф -1
+    int count = 0;
+    for (int i = 0; i < strlen(str); i++) {
+        if (str[i] == '(') {
+            count++;
+
+        } else if (str[i] == ')') {
+            count--;
+        }
+    }
+
+    // если после пробежки по строке коэф неравен 0, значит, количество открывающих и закрывающих скобок разное
+    if (count == 0) {
+        return true;
+    } else {
+        return false;
+    }
```

Comment:
```suggestion
    return count == 0;
```
Вообще экивалентно этому. Запомните и не пишите так больше.